### PR TITLE
BREAKING: feat: make lambda pick up the script from the request body

### PR DESCRIPTION
**Pull request by NamanJain8**
Earlier, dgraph lambda used to use a predefined lambda script from the file. This poses few limitations on how `lambda` are handled in multi-tenant environment.

This change makes `dgraph-lambda` accept `source` field in the request body. This field will contain the required js script, evaluate it at runtime and execute the query.

### CODE REVIEW

File: src/index.test.ts [hunk 1]
```diff
@@ -1,28 +1,30 @@
-import { scriptToExpress } from "./script-to-express";
+import { buildApp } from "./index";
 import supertest from 'supertest'
 
-describe(scriptToExpress, () => {
+describe(buildApp, () => {
+  const app = buildApp()
```
Comment (tstreamDOTh) [inline src/index.test.ts:5]: For the teardown issue try adding `afterEach` and call `app.close()`

**Review by tstreamDOTh — APPROVED** [review]

**Comment by gja** [comment]
Heya @NamanJain8 @manishrjain. 

Saw this PR. Something you might want to consider is that with this change, it would be technically possible for anyone to effectively shut down all lambda servers by submitting a simple for loop in the lambda code, unless you control this out of band (say by isolating different processes)

**Comment by amaster507** [comment]
@gja I have been watching this too. I think the direction is good, but maybe a timeout on the script would fix that concern? But still curious what would happen if more than one script was running back to back. Would the second script have to wait on the first, or will the scripts run async? (I understand async still has to wait for a pause in processing script 1 before it can start working on script 2 and then another pause to come back and finish up working on script 1.)

**Comment by gja** [comment]
I've sent the security vulnerability over to manish and naman. Very happy
to chat with you over email.

**Comment by tstreamDOTh** [comment]
Adding a middleware for timeout in express should solve the problem @NamanJain8

**Comment by tstreamDOTh** [comment]
For namespace specific tenant logging- you extend `console` object or use `proxy`

File: src/evaluate-script.ts [hunk 2]
```diff
@@ -43,7 +43,25 @@ class GraphQLResolverEventTarget extends EventTarget {
...
+}
+
+function newContext(eventTarget: GraphQLResolverEventTarget, logPrefix: string) {
+  // Override the console object to append prefix in front.
+  console.debug = appendPrefix(console.debug, logPrefix)
+  console.error = appendPrefix(console.error, logPrefix)
+  console.info = appendPrefix(console.info, logPrefix)
+  console.log = appendPrefix(console.log, logPrefix)
+  console.warn = appendPrefix(console.warn, logPrefix)
+
```
Comment (tstreamDOTh) [inline src/evaluate-script.ts:64]: Lets add unit tests for these

**Pull request closed without merging**